crates/core/src/util/lending_pool.rs

```diff
@@ -52,50 +49,38 @@ struct PoolVec<T> {
 #[derive(Debug)]
 pub struct PoolClosed;
 
-impl<T, G: WaiterGauge> LendingPool<T, G> {
-    pub fn new() -> Self
-    where
-        G: Default,
-    {
-        Self::with_gauge(G::default())
-    }
-
-    pub fn with_gauge(waiter_gauge: G) -> Self {
-        Self::from_iter_with_gauge(std::iter::empty(), waiter_gauge)
-    }
-
-    pub fn from_iter_with_gauge<I: IntoIterator<Item = T>>(iter: I, waiter_gauge: G) -> Self {
-        let deque = VecDeque::from_iter(iter);
-        Self {
-            sem: Arc::new(Semaphore::new(deque.len())),
-            waiter_gauge,
-            inner: Arc::new(LendingPoolInner {
-                closed_notify: NotifyOnce::new(),
-                vec: Mutex::new(PoolVec {
-                    total_count: deque.len(),
-                    deque: Some(deque),
-                }),
-            }),
-        }
-    }
-
-    pub fn request(&self) -> impl Future<Output = Result<LentResource<T>, PoolClosed>>
-    where
-        G: for<'a> WaiterGauge<Context<'a> = ()>,
-    {
-        self.request_with_context(())
+impl<T> LendingPool<T> {
+    pub fn new() -> Self {
+        Self::from_iter(std::iter::empty())
     }
 
     pub fn request_with_context(
         &self,
-        context: G::Context<'_>,
+        (identity, module_hash, database_address, reducer_symbol): (Identity, Hash, Address, &str),
     ) -> impl Future<Output = Result<LentResource<T>, PoolClosed>> {
         let acq = self.sem.clone().acquire_owned();
         let pool_inner = self.inner.clone();
-        let waiter_guard = self.waiter_gauge.inc(context);
+
+        let queue_len = WORKER_METRICS.instance_queue_length.with_label_values(
```

Comment:
this is fine, I guess. it just makes it less easy to pull out/appraise as its own data structure. I might take a look at making this much simpler while still keeping LendingPool generic.